### DISCUSSION TITLE
Consult previews don't have any font highlighting

### DISCUSSION THREAD

**Discussion by sudo-human** [minad/consult#1002] [Q&A] [answered]
The preview buffer for consult commands no longer show syntax highlighting and line numbers. I've tested this barebone emacs as given in the readme of this package. Here are the screenshots:

The visited buffer shows syntax highlighting
<img width="1222" alt="Screenshot 2024-04-18 at 9 11 53 PM" src="https://github.com/minad/consult/assets/13881173/9a2ff9ca-3b10-4b5c-8d54-a77b186f6509">

The preview buffer of the other file doesn't show any syntax highlighting or line numbers
<img width="1222" alt="Screenshot 2024-04-18 at 9 07 45 PM" src="https://github.com/minad/consult/assets/13881173/aa5e66e8-6d25-4a37-821d-9877e9aac086">

I'm using the latest emacs from `emacs-plus` formula for mac:
`GNU Emacs 30.0.50 (build 1, aarch64-apple-darwin23.4.0, NS appkit-2487.50 Version 14.4.1 (Build 23E224)) of 2024-04-18`

**Comment by minad** [comment]
Do you also see this issue on Emacs 29? I am not using Emacs 30 yet, since it is unstable and unreliable. It could be that there is a regression.

**Reply by sudo-human** [reply]
On emacs 29 I've seen that the line numbers disappear. I'm using `global-display-line-numbers-mode` for line numbers. But syntax highlighting is there.

**Reply by minad** [reply]
> On emacs 29 I've seen that the line numbers disappear. I'm using global-display-line-numbers-mode for line numbers. But syntax highlighting is there.

This is expected. You can adjust consult-preview-allowed-hooks to include the line numbers hook used by global-display-line-numbers-mode. Font-locking is also enabled that way. It may be that there has been a change in this infrastructure in Emacs 30.

**Reply by sudo-human** [reply]
I tried looking through the display-line-numbers code and the only hook I found out was `display-line-numbers-update-width` from the below line
`(add-hook 'pre-command-hook #'display-line-numbers-update-width nil t))`

Will adding `display-line-numbers-update-width` to `consult-preview-allowed-hooks` solve the issue? And anything similar that should I do for font-locking?

**Reply by minad** [reply]
To enable line numbers during preview, this should work as of 687f7e4c4ef02ed26c9a5d2718390e91b276a388:

~~~ elisp
(add-to-list 'consult-preview-allowed-hooks 'global-display-line-numbers-mode)
~~~

**Comment by minad** [comment, marked as answer]
Fixed in https://github.com/minad/consult/commit/687f7e4c4ef02ed26c9a5d2718390e91b276a388. Emacs 30 contained a backward incompatible change.
